cacheops/redis.py

```diff
@@ -18,13 +20,27 @@ def handle_connection_failure(call):
             warnings.warn("The cacheops cache is unreachable! Error: %s" % e, RuntimeWarning)
         except redis.TimeoutError as e:
             warnings.warn("The cacheops cache timed out! Error: %s" % e, RuntimeWarning)
+        except Exception as e:
+            warnings.warn(traceback.format_exception(*sys.exc_info()))
 else:
     handle_connection_failure = identity
 
 
 class SafeRedis(redis.StrictRedis):
     get = handle_connection_failure(redis.StrictRedis.get)
 
+    """ Handles failover of AWS elasticache
+    """
+    def execute_command(self, *args, **options):
+        try:
+            return super(SafeRedis, self).execute_command(*args, **options)
+        except redis.ResponseError as e:
+            if "READONLY" not in e.message:
+                raise
+            connection = self.connection_pool.get_connection(args[0], **options)
+            connection.disconnect()
+            warnings.warn("Primary probably failed over, reconnecting")
+            return super(SafeRedis, self).execute_command(*args, **options)
```

Comment:
To answer your question, who knows but this code accounts for it so it shouldn't matter :)